src/python/pants/engine/engine_aware.py

```diff
@@ -56,3 +56,9 @@ def artifacts(self) -> Optional[Dict[str, Snapshot]]:
         `artifacts` is a mapping of arbitrary string keys to `Snapshot`s.
         """
         return None
+
+    def metadata(self) -> Optional[Dict[str, str]]:
```

Comment:
we should allow for more flexibility in the values.
i.e. allow str, int, lists of str, ints.